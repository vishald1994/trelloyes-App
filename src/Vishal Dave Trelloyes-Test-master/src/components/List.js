import React from 'react';
import '../styles/List.css';

import Card from './Card';


class List extends React.Component {

  render() {
    const cards = this.props.cards.map(card => {
      return <Card
        key={card.id}
        title={card.title}
        content={card.content}
      />
    });
    
    return (
      <section className="List">
        <header className="List-header">
          <h2>{this.props.header}</h2>
        </header>
        <div className="List-cards">
          {cards}
        </div>
      </section>
    )
  }
}

export default List;