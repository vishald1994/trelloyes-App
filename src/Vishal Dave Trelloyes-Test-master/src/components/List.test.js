import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import List from './List';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<List
    key='0'
    header="Test Header"
    cards={[]}
    />, div);

  ReactDOM.unmountComponentAtNode(div);    
});

it ('renders this UI as expected', () => {
  const app = renderer.create(<List key='0' header='test header' cards={[]}/>).toJSON();
  expect(app).toMatchSnapshot();
})