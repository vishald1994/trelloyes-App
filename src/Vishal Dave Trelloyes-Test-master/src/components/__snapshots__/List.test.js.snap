// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders this UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      test header
    </h2>
  </header>
  <div
    className="List-cards"
  />
</section>
`;
